### ISSUE TITLE
Invalid symbolic link ~/.cache/pex/unzipped_pexes/<hash>/.bootstrap on Fedora Silverblue/Kinoite 41

### ISSUE THREAD

**Issue by tymonx** [pex-tool/pex#2634]
The [pants](https://github.com/pantsbuild/pants) project uses pex and because of that it doesn't work on Linux distributions like Fedora Kinote or Silverblue (Atomic OS). Because of `~/` or `/home/${USER}/` that is a symbolic link to `/var/home/${USER}/`. `pex` creates a symbolic link `~/.cache/pex/unzipped_pexes/<hash>/.bootstrap` pointing to `../../../../../../var/home/${USER}/.cache/pex/bootstraps/<hash>`. The main root cause is relative content of that created symbolic link. If `pex` file is run when current working directory is part of `/home/${USER}/` then unpacking procedure fails with raised Python exception from `~/.cache/pex/unzipped_pexes/<hash>/__main__.py` file:

```plaintext
from pex.globals import Globals

ModuleNotFoundError: No module named 'pex'
```

Because of invalid symbolic link added to `sys.path[0]` by above `__main__.py` file, the Python `pex` module cannot be found by Python interpreter.

It impacts [pants](https://github.com/pantsbuild/pants): https://github.com/pantsbuild/pants/issues/21799

Related issues: https://github.com/pex-tool/pex/issues/2572 and https://github.com/pex-tool/pex/pull/2626 Unfortunately it doesn't fix above issue for pants `2.31.1`.

It looks like that fix should be fair easy to implement (using `realpath`).

**Comment by jsirois** [comment]
@tymonx how did you determine  #2572 and #2626 do not fix the issue. Have you verified Pants and scie-pants use versions of Pex incorporating both of those fixes?

**Comment by jsirois** [comment]
For example #2626 was only released in Pex 2.27.1 2 weeks ago: https://github.com/pex-tool/pex/releases/tag/v2.27.1

Surely the Pants folks haven't updated Pants and scie-pants to that already.

**Comment by jsirois** [comment]
@tymonx this was in fact fixed in #2626 but you need Pants 2.25.0.dev3 or newer to get that fix.

See the example docker session here: https://github.com/pantsbuild/pants/issues/21799#issuecomment-2563099162

**Comment by jsirois** [comment]
@tymonx I'm going to close this as an answered question since https://github.com/pantsbuild/pants/issues/21799#issuecomment-2563099162 proves the fixes in https://github.com/pex-tool/pex/issues/2572 and https://github.com/pex-tool/pex/pull/2626 do work. You just may have to wait awhile on Pants to get a stable 2.25 release out if you're not willing to use a Pants 2.25 dev release.

**Comment by tymonx** [comment]
@jsirois You are right. It uses older version of `pex` `2.16.2` before fix. I will try `Pants 2.25.dev3`:

```plaintext
cat ~/.cache/pex/bootstraps/<hash>/pex/version.py
```

**Comment by tymonx** [comment]
@jsirois I can confirm. If fixes the issue when changing to version `2.25.dev3` in `pants.toml` file:

```toml
[GLOBAL]
pants_version = "2.25.0.dev3"
```

**Comment by jsirois** [comment]
Great - thanks for confirming @tymonx.
